src/darwin/Framework/CHIP/MTRBaseDevice.mm

```diff
@@ -147,9 +147,13 @@ static void PurgeReadClientContainers(
                     container.readClientPtr = nullptr;
                 }
                 if (container.pathParams) {
-                    Platform::Delete(container.pathParams);
+                    Platform::MemoryFree(container.pathParams);
```

Comment:
This assumes that AttributePathParams has a trivial destructor.  We should be static_asserting that, so if the assumption stops being true we can fix this code accordingly.  Something like:
```suggestion
                    static_assert(std::is_trivially_destructible<AttributePathParams>::value, "AttributePathPArams destructors won't get run");
                    Platform::MemoryFree(container.pathParams);
```